exec.go

```diff
@@ -111,6 +108,9 @@ func (db *DB) Insert(table string, data interface{}) error {
 }
 
 func setPrimaryKey(rv reflect.Value, id int64) {
+	if !rv.CanAddr() {
```

Comment:
I am not in favor of checking CanAddr in setPrimaryKey, it hides logic and leaves the caller with a silent discard. It is better to put that responsibility into the callers hands, especially since there are already other checks performed before calling setPrimaryKey. So that would need to be changed.

Reply:
I have moved the check out of the function and added it before calling setPrimaryKey. I am not completely in favor of this, since we need to check every time before calling setPrimaryKey, resulting in higher computation time.